internal/kafka/internal/handlers/kafka.go

```diff
@@ -42,10 +47,34 @@ func (h kafkaHandler) Create(w http.ResponseWriter, r *http.Request) {
 			handlers.ValidateAsyncEnabled(r, "creating kafka requests"),
 			handlers.ValidateLength(&kafkaRequest.Name, "name", &handlers.MinRequiredFieldLength, &MaxKafkaNameLength),
 			ValidKafkaClusterName(&kafkaRequest.Name, "name"),
+			// ValidatePlan(h.service, h.kafkaConfig, &kafkaRequest.Plan, convKafka),
 			ValidateKafkaClusterNameIsUnique(&kafkaRequest.Name, h.service, r.Context()),
 			ValidateKafkaClaims(ctx, &kafkaRequest, convKafka),
 			ValidateCloudProvider(&h.service, convKafka, h.providerConfig, "creating kafka requests"),
 			handlers.ValidateMultiAZEnabled(&kafkaRequest.MultiAz, "creating kafka requests"),
+			func() *errors.ServiceError { // Validate plan
+				instanceType, err := h.service.DetectInstanceType(convKafka)
+				if err != nil {
+					return err
+				}
+				if !stringNotSet(&kafkaRequest.Plan) {
+					p := strings.Split(kafkaRequest.Plan, ".")
+					_, e := h.kafkaConfig.GetKafkaInstanceSize(instanceType.String(), p[1])
```

Comment:
there might not be a `p[1]` leading to a nil pointer exception. Better check for array length before this.

Reply:
If our standard will be that `plan` will always be in form of `instanceType.sizeid`, I believe it would be better to create a `Plan` type with two methods: `GetInstanceType` and `GetSizeId`

```go
type Plan string

func(p Plan) GetInstanceType() {
	...
}

func (p Plan) GetSizeID() {
	...
}
```